src/toolkits/object_detection/od_data_iterator.cpp

```diff
@@ -103,6 +103,10 @@ std::vector<image_annotation> parse_annotations(
 
       if (key == "label") {
         // If the label is invalid (not in class_to_index_map) then ignore it.
+        if (kv.second.get_type() == flex_type_enum::UNDEFINED) {
+          log_and_throw("Annotation labels must have valid type.");
```

Comment:
add more informational text like: please consider 'dropna()' to clean column <blah>. Also, Check #2804 to early error out user the missing value will halt the training.

Reply:
good!

Reply:
will not do the same thing in #2804 in `init_training` tho,
bc OD allows empty annotation (they ignore them and this is covered in a test case),
and the missing value for image is handled in the Python side.